tools/cmake/cores/Cortex-M7F.cmake

```diff
@@ -16,10 +16,10 @@ elseif(${MBED_TOOLCHAIN} STREQUAL "ARM")
         "-mfloat-abi=hard"
     )
     list(APPEND asm_compile_options
-        "-mcpu=Cortex-M7.fp.sp"
+        "-mcpu=Cortex-M7"
```

Comment:
I think the problem here is that you're using `armclang` to invoke the assembler, but passing it armasm-type CPU options.

I believe your `asm_compile_options` should be the same as the `c_cxx_compile_options`, including the `-mfpu` and `armclang` should translate.

Not sure about the link, but potentially the same thing applies if you're invoking it via `armclang`. In which case you're in the same situation as GCC, and they should all be `common_options`.

Reply:
> I think the problem here is that you're using armclang to invoke the assembler (why?), but passing it armasm-type CPU options.

I've missed this earlier. We were suggested due to another bug to use armclang and --auto mode. The flags should be fixed m thanks for noticing.